sdk/storage/AzureStorageBlob/Source/Data/Transfer.swift

```diff
@@ -30,11 +30,14 @@ import Foundation
 // MARK: Protocols
 
 public protocol Transfer: AnyObject {
-    var rawState: Int16 { get set }
+    var id: UUID { get }
     var state: TransferState { get set }
-    var operation: ResumableOperation? { get set }
     var debugString: String { get }
-    var id: UUID { get }
+    var rawState: Int16 { get set }
+}
+
+internal protocol TransferImpl: Transfer {
```

Comment:
Since protocols can't define access levels for members, and the `Transfer` protocol needs to be public, but I didn't want implementation details to leak into the public protocol, I made this internal protocol that extends the public one. Public API methods can expose `Transfer` in their signature, but internally they can work with `TransferImpl` objects to access the extra properties that are meant to be internal-only.

Unfortunately I wanted to but was unable to hide `rawState` here as well, because it's used by the getter/setter for `state`, which has to be in the public `Transfer` protocol.